components/display/ili9341.rst

```diff
@@ -0,0 +1,123 @@
+ILI9341 TFT LCD
+===============
+
+.. seo::
+    :description: Instructions for setting up ILI9341 TFT LCD display drivers.
+    :image: 24-ili9341-full.jpg
+
+.. _ili9341:
+
+Usage
+-----
+
+The ``ili9341`` display platform allows you to use
+ILI9341 (`datasheet <https://cdn-shop.adafruit.com/datasheets/ILI9341.pdf>`__,
+`Ali model used <https://s.click.aliexpress.com/e/_dTKH6Mt>`__)
+displays with ESPHome. Note that this display controller uses a lot of pins even if touch
+is not used. So it's better to used it with an EPS32 though esp8266 might be suitable
+too (with some SPI flash changes). 
+
+.. figure:: images/24-ili9341-full.jpg
+    :align: center
+    :width: 75.0%
+
+    ILI9341 common display
+
+
+.. code-block:: yaml
+
+    # Example minimal configuration entry
+    display:
+      - platform: ili9341
+        model: TFT_2.4
+        cs_pin: 14
+        dc_pin: 27
+        # led_pin: 32 // you may use it with a constant brightness but there is a better way
+        reset_pin: 33
+        rotation: 0
+        
+        lambda: |-
+          it.fill(ili9341::BLACK);
+          it.print(0, 0, id(font), "Hello World!");
+    
+Configuration variables
+***********************
+
+- **led_pin** (:ref:`Pin Schema <config-pin_schema>`): The display's backlight pin.
+- **cs_pin** (:ref:`Pin Schema <config-pin_schema>`): The CS pin.
+- **dc_pin** (:ref:`Pin Schema <config-pin_schema>`): The DC pin.
+- **reset_pin** (:ref:`Pin Schema <config-pin_schema>`): The RESET pin.
```

Comment:
mark this optional, as it is in the schema, also I was testing this and found out that the thing does not work if the pin is not connected (do you know why?)

Reply:
Hi there, what pin are you talking about? reset_pin?

Reply:
sorry, I messed up there, anyway, all pins should be marked either optional or required following the schema definition rules.

Reply:
also, the `reset_pin` it is marked optional on the schema, how is it supposed to work if it is not connected?